proarc-webapp/src/main/java/cz/cas/lib/proarc/webapp/client/widget/workflow/WorkflowJobView.java

```diff
@@ -265,6 +288,83 @@ private void fetchAddSubjobMenu(Record job) {
         });
     }
 
+    private void fetchModelMenu(Record job) {
+        if (job == null
+                // not subjob
+                || job.getAttribute(WorkflowJobDataSource.FIELD_PARENTID) != null
+                || !Job.State.OPEN.name().equals(job.getAttribute(WorkflowJobDataSource.FIELD_STATE))
+                ) {
+            createNewObjectButton.setVisible(false);
+            return;
+        }
+        String jobName = job.getAttribute(WorkflowJobDataSource.FIELD_PROFILE_ID);
+        Long jobId = job.getAttributeAsLong(WorkflowJobDataSource.FIELD_ID);
+
+        if (jobName == null) {
+            return;
+        }
+
+        Criteria criteria = new Criteria();
+        criteria.setAttribute(WorkflowMaterialDataSource.FIELD_JOB_ID, jobId);
+        criteria.setAttribute(WorkflowMaterialDataSource.FIELD_TYPE, MaterialType.DIGITAL_OBJECT);
+        WorkflowMaterialDataSource.getInstance().fetchData(criteria, (dsResponse, o, dsRequest) -> {
+            RecordList records = dsResponse.getDataAsRecordList();
+            if (records.getLength() == 1 && records.get(0)
+                    .getAttribute(WorkflowMaterialDataSource.FIELD_DIGITAL_PID) == null) {
+                createNewObjectButton.enable();
+            } else {
+                createNewObjectButton.disable();
+            }
+        });
+
+
+        WorkflowProfileDataSource.getInstance().getModels(false, jobName, (models) -> {
+            Menu menu = new Menu();
+
+            for (Record model : models) {
+                MenuItem menuItem = new MenuItem(model.getAttribute(WorkflowProfileConsts.MODEL_TITLE));
+                menu.addItem(menuItem);
+
+                menuItem.addClickHandler(event -> {
+                    saveNewDigitalObject(model.getAttributeAsString(WorkflowProfileConsts.MODEL_NAME), jobId);
+                    createNewObjectButton.disable();
+                    actionSource.fireEvent();
+                    jobFormView.refreshState();
+                    jobFormView.refresh();
+                });
+
+            }
+
+            createNewObjectButton.setVisible(models.length > 0);
+            createNewObjectButton.setMenu(menu);
+        });
+    }
+
+    private void saveNewDigitalObject(String modelId, Long jobId) {
```

Comment:
Možnost PIDu ve specifikaci není a ani to nemám do GUI kam dát.

![img](http://i.imgur.com/BLjPNrI.gif)

Objekt je založen ihned po kliknutí a potom se zobrazí dialogové okno.